msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMSALController.java

```diff
@@ -169,16 +177,51 @@ private Map getSilentParameters(MSALAcquireTokenSilentOperationParameters parame
         request.setForceRefresh(parameters.getForceRefresh());
         request.setLoginHint(parameters.getAccount().getUsername());
         request.setName(parameters.getAccount().getUsername());
-        request.setUserId(parameters.getAccount().getHomeAccountId());
+        request.setUserId(parameters.getAccount().getLocalAccountId());
         //request.setPrompt(parameters.get);
         //TODO: This should be the broker redirect URI and not the non-broker redirect URI
         request.setRedirect(parameters.getRedirectUri());
         //NOTE: Assumption: Broker will handle removing empty string scopes and appending stanard OIDC scopes
         request.setScope(StringUtil.join(' ', parameters.getScopes()));
+        request.setApplicationName(parameters.getAppContext().getPackageName());
         //request.setVersion();
 
+        return request;
+    }
+
+    private AuthenticationResult getAuthenticationResult(BrokerTokenResponse brokerTokenResponse){
+        try {
+            ClientInfo clientInfo = new ClientInfo(brokerTokenResponse.getClientInfo());
+            String homeAccountId = SchemaUtil.getHomeAccountId(clientInfo);
+            String idToken = brokerTokenResponse.getIdToken();
+            String tenantId = clientInfo.getUtid();
+
+            AccessTokenRecord accessTokenRecord = new AccessTokenRecord();
```

Comment:
Do we need to set the `local_account_id` here? I suspect **YES**: this value is surfaced to users through `IAccount#getAccountIdentifier` in `AuthenticationResult#getAccount`

Reply:
`AccessTokenRecord` doesn't have a local_account_id right?

Reply:
Just kidding ignore me :-)